WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift

```diff
@@ -103,26 +107,44 @@ final class AnalyticsHubViewModel: ObservableObject {
 private extension AnalyticsHubViewModel {
 
     @MainActor
-    func retrieveOrderStats() async throws {
+    func retrieveData() async throws {
         switchToLoadingState()
 
         let currentTimeRange = try timeRangeSelection.unwrapCurrentTimeRange()
         let previousTimeRange = try timeRangeSelection.unwrapPreviousTimeRange()
 
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+                try await self.retrieveOrderStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
+            }
+            group.addTask {
+                try await self.retrieveVisitorStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
+            }
+            try await group.waitForAll()
+        }
+    }
```

Comment:
Newbie question: Could we use a regular `async let` to keep requests more consistent, from reading at the code it is not clear to me why we need the task group, I'm sure we are missing something 😅

```swift
async let request1: () = retrieveOrderStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
async let request2: () = retrieveVisitorStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)

_ = try await(request1, request2)
```

Reply:
Good question! I agree that simplest case of "start parallel requests and don't manage them at all" looks pretty complicated in async/await.

Your suggestion was one of my attempts, but in my testing it stopped all parallel requests when only one of them errored. I'll add a couple of unit tests for edge cases I've noticed.